src/hotspot/os/linux/os_linux.cpp

```diff
@@ -4821,6 +4822,20 @@ jint os::init_2(void) {
     }
   }
 
+  // Check the availability of MADV_POPULATE_WRITE.
+  if (FLAG_IS_DEFAULT(UseMadvPopulateWrite) && UseMadvPopulateWrite) {
```

Comment:
Should we really allow the user to forcefully set this and trigger a path that is not going to be correct on old kernels?